### ISSUE TITLE
String with more than one `count`

### ISSUE THREAD

**Issue by victornpb** [i18next/i18next#908]
Today I encountered a situation that need to have a string with more than one `count`, and I was not sure how to deal with it.

```
i18next.t('foo', { count_cars: x, count_persons: y });
``` 

```
{
     "foo": "There are {count_cars} car(s) with {count_persons} person(s) in it."
}
```
How would I structure this to offer the 4 possible plural/singular combinations?

**Comment by jamuhl** [comment]
should be possible like: http://i18next.com/translate/nesting/#withpluralandoptions

**Comment by jamuhl** [comment]
did this solve your demand?

**Comment by victornpb** [comment]
yes, not as clean as I wish, but it does.

**Comment by jamuhl** [comment]
yes...not the easiest thing to reach...looking at languages having multiple pluralforms that stuff gets complicated very fast. If you got an idea to improve that?

**Comment by victornpb** [comment]
I thought we can have something like this.

```
t('foo.msg', {girl: 2, boy: 1}) // "They have 2 girls and 1 boy."
```


```
function t(key, options){
	var string = i18next.t(key, options);
	string = string.replace(/\{\{\+(\w+)\}\}/g, function(m, token){
		return i18next.t(key+"_"+token, {count: token});
	});
	return string;
}
```



Your locale files would look like this.
```
{
	"foo": {
		"msg": "They have {{+girl}} and {{+boy}}.",
		"msg_girl": "{{count}} girl",
		"msg_girl_plural": "{{count}} girls",
		"msg_boy": "{{count}} boy",
		"msg_boy_plural": "{{count}} boys"
	}
}
```
or this will also works
```
{
	"foo": {
		"msg": "They have {{girl}} {{+girl}} and {{boy}} {{+boy}}.",
		"msg_girl": "girl",
		"msg_girl_plural": "girls",
		"msg_boy": "boy",
		"msg_boy_plural": "boys"
	}
}
```

Let me know if you like the concept.

**Comment by jamuhl** [comment]
hm...rather inspiring, what might work is:

```
{
	"foo": {
		"msg": "They have {{girl}} ${girl} and {{boy}} ${boy}.",
		"msg_girl": "girl",
		"msg_girl_plural": "girls",
		"msg_boy": "boy",
		"msg_boy_plural": "boys"
	}
}
```

which is a 'regular' nesting call, but here:

https://github.com/i18next/i18next/blob/master/src/Interpolator.js#L126

`value = fc(handleHasOptions.call(this, match[1].trim()), clonedOptions);`

we call basically i18next.t with new cloned options there we might transform the girl/boy count value into a regular count for the nested call (`match[1].trim()` is girl or boy).

```
// eventual we should add a flag for this behaviour to keep it backwards compatible?!?
t('foo.msg', {girl: 2, boy: 1, transformCounts: true })
```

**Comment by victornpb** [comment]
Can I make it a contribution?

Oh and I see lots of things that I would argue about the nesting operator. I have feelings that it should be deprecated on how it is now, but I don't have enough background on the motivation to have this feature to begin with. I have a strong feeling that nesting with options shouldn't be a thing, I can see it being useful in some circumstances, but at least shouldn't be encouraged.

**Comment by jamuhl** [comment]
Agree on nesting with options is somehow a little over the top...was introduced in v1. But as it's not overly complicated i would keep it.

But i think we could add the convert for "girl" to count without the need of transformCounts - if new options included girl and count i think it's save and backwards compatible.

If you like you can make the contribution. PR would be welcome...else i will see if i got on this (hopefully sooner than later)

**Comment by jamuhl** [comment]
argh...missed that one...will reopen for implementation...hopefully get this done soon...as it looks like a nice improvement

**Comment by victornpb** [comment]
Oh, I started implementing this to submit it as a PR, but I got stuck at some point, I'm not familiar with the code base and then I totally forgot about it.

**Comment by jamuhl** [comment]
didn't got to it myself...so still hanging here for a long time :( but lets keep it in the hope xmas holiday gives the time

**Comment by jamuhl** [comment]
guess we will close this for now...PR would still be welcome but i guess for the few cases this is needed the current solution is an option.

If more often needed i guess using ICU would be easier: https://github.com/i18next/i18next-icu

**Comment by andreialecu** [comment]
I stumbled upon this issue while looking for similar functionality.

Almost switched from `i18next` format because of this issue and was thinking of moving to `icu` or `fluent`.

This seems to be now be possible via https://www.i18next.com/translation-function/nesting#passing-options-to-nestings

The example in the docs works, but I find this clearer:

```json
{
      "girlsAndBoys": "They have $t(girls, {\"count\": {{girls}} }) and $t(boys, {\"count\": {{boys}} })",
      "boys": "{{count}} boy",
      "boys_plural": "{{count}} boys",
      "girls": "{{count}} girl",
      "girls_plural": "{{count}} girls"
}
```

```ts
i18next.t('girlsAndBoys', {girls: 3, boys: 2}); // They have 3 girls and 2 boys 
```

**Comment by ScreamZ** [comment]
@andreialecu hello :) the world is small mate

**Comment by sankaSanjeeva** [comment]
> i18next.t('girlsAndBoys', {girls: 3, boys: 2});

I did something like this and got this output
`i18next.t('myNameSpace:someKey.girlsAndBoys', {girls: 3, boys: 2}); // They have girls and boys `

So I understood translations should be like this

```
{
      "girlsAndBoys": "They have $t(myNameSpace:someKey.girls, {\"count\": {{girls}} }) and $t(myNameSpace:someKey.boys, {\"count\": {{boys}} })",
      "boys": "{{count}} boy",
      "boys_plural": "{{count}} boys",
      "girls": "{{count}} girl",
      "girls_plural": "{{count}} girls"
}
```
